scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -690,6 +696,128 @@ impl ReplicatorConsumer {
         Ok(())
     }
 
+    fn delete_row_range_left(&mut self, mut data: CDCRow<'_>, included: bool) {
+        self.left_range_included = included;
+        self.left_range_values =
+            take_clustering_keys_values(&self.source_table_data.table_schema, &mut data);
+    }
+
+    async fn delete_row_range_right(
+        &mut self,
+        data: CDCRow<'_>,
+        right_included: bool,
+    ) -> anyhow::Result<()> {
+        let table_schema = &self.source_table_data.table_schema;
+        if table_schema.clustering_key.is_empty() {
+            return Ok(());
+        }
+
+        let values_left = std::mem::take(&mut self.left_range_values);
+        let values_left = values_left.iter().map(|x| x.as_ref()).collect::<Vec<_>>();
+
+        let (_, timestamp, values) = self.get_common_cdc_row_data(&data);
+
+        let values_right = &values[table_schema.partition_key.len()..];
+
+        let first_unequal_position = values_left
+            .iter()
+            .zip(values_right.iter())
+            .position(|pair| pair.0 != pair.1)
+            .unwrap_or(table_schema.clustering_key.len());
+
+        let keys_equality_cond = table_schema
+            .partition_key
+            .iter()
+            .map(|name| format!("{} = ?", name))
+            .join(" AND ");
+
+        let mut conditions = vec![keys_equality_cond];
+
+        let mut query_values = values[..table_schema.partition_key.len()].to_vec();
+
+        let less_than = if right_included { "<=" } else { "<" };
+        let greater_than = if self.left_range_included { ">=" } else { ">" };
+
+        self.add_range_condition(
+            &mut conditions,
+            &mut query_values,
+            &values_left,
+            first_unequal_position,
+            greater_than,
+        );
+        self.add_range_condition(
+            &mut conditions,
+            &mut query_values,
+            values_right,
+            first_unequal_position,
+            less_than,
+        );
+
+        let query = Query::new(format!(
+            "DELETE FROM {}.{} WHERE {}",
+            self.precomputed_queries
+                .destination_table_params
+                .dest_keyspace_name,
+            self.precomputed_queries
+                .destination_table_params
+                .dest_table_name,
+            conditions.join(" AND ")
+        ));
+
+        run_statement(
+            &self
+                .precomputed_queries
+                .destination_table_params
+                .dest_session,
+            query,
+            &query_values,
+            timestamp,
+        )
+        .await
+    }
+
+    fn add_range_condition<'a>(
+        &self,
+        conditions: &mut Vec<String>,
+        query_values: &mut Vec<Option<&'a CqlValue>>,
+        values: &[Option<&'a CqlValue>],
+        first_unequal_position: usize,
+        relation: &str,
+    ) {
+        if values[first_unequal_position].is_some() {
```

Comment:
Is this correct? How does this work e.g. for `ck1 = 1 AND ck2 < 2`? Won't this accidentally skip the lower bound? The correct conditions should be `(ck1) < (1) AND (ck1, ck2) > (1, 2)`.

I added some debug prints to the replicator logic and it seems that it generates a wrong query for the second delete in the `test_range_delete`, but the test accepts it although the incorrect delete should delete more rows than necessary. I wonder what's wrong (with the code or my understanding).